examples/name-service-without-macros/programs/name-service/src/lib.rs

```diff
@@ -0,0 +1,222 @@
+        let account_infos = convert_metas_to_infos(&inputs.accounts, &crate::ID)?;
+        let light_accounts = LightDeleteRecord::try_light_accounts(&account_infos)?;
+
+        if light_accounts.record.owner != ctx.accounts.signer.key() {
+            return err!(CustomError::Unauthorized);
+        }
+
+        verify_compressed_accounts(
+            &ctx,
+            inputs.proof,
+            &[light_accounts.record],
+            None,
+            false,
+            None,
+            &crate::ID,
+        )?;
+
+        Ok(())
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, BorshDeserialize, BorshSerialize)]
+pub enum RData {
+    A(Ipv4Addr),
+    AAAA(Ipv6Addr),
+    CName(String),
+}
+
+impl anchor_lang::IdlBuild for RData {}
+
+impl AsByteVec for RData {
+    fn as_byte_vec(&self) -> Vec<Vec<u8>> {
+        match self {
+            Self::A(ipv4_addr) => vec![ipv4_addr.octets().to_vec()],
+            Self::AAAA(ipv6_addr) => vec![ipv6_addr.octets().to_vec()],
+            Self::CName(cname) => cname.as_byte_vec(),
+        }
+    }
+}
+
+impl Default for RData {
+    fn default() -> Self {
+        Self::A(Ipv4Addr::new(127, 0, 0, 1))
+    }
+}
+
+#[derive(
+    Clone, Debug, Default, AnchorDeserialize, AnchorSerialize, LightDiscriminator, LightHasher,
+)]
+pub struct NameRecord {
+    #[truncate]
+    pub owner: Pubkey,
+    #[truncate]
+    pub name: String,
+    pub rdata: RData,
+}
+
+#[error_code]
+pub enum CustomError {
+    #[msg("No authority to perform this action")]
+    Unauthorized,
+}
+
+#[light_system_accounts]
+#[derive(Accounts, LightTraits)]
+pub struct CreateRecord<'info> {
+    #[account(mut)]
+    #[fee_payer]
+    pub signer: Signer<'info>,
+    #[self_program]
+    pub self_program: Program<'info, crate::program::NameService>,
+    /// CHECK: Checked in light-system-program.
+    #[authority]
+    pub cpi_signer: AccountInfo<'info>,
+}
+
+pub struct LightCreateRecord<'a> {
+    pub record: LightAccount<'a, NameRecord>,
+}
+
+impl<'a> LightAccounts<'a> for LightCreateRecord<'a> {
+    fn try_light_accounts(accounts: &'a [LightAccountInfo]) -> Result<Self> {
+        let record: LightAccount<NameRecord> = LightAccount::new(&accounts[0])?;
+        Ok(Self { record })
+    }
+}
+
+#[light_system_accounts]
+#[derive(Accounts, LightTraits)]
+pub struct UpdateRecord<'info> {
+    #[account(mut)]
+    #[fee_payer]
+    pub signer: Signer<'info>,
+    #[self_program]
+    pub self_program: Program<'info, crate::program::NameService>,
+    /// CHECK: Checked in light-system-program.
+    #[authority]
+    pub cpi_signer: AccountInfo<'info>,
+}
+
+pub struct LightUpdateRecord<'a> {
+    pub record: LightAccount<'a, NameRecord>,
+}
+
+impl<'a> LightAccounts<'a> for LightUpdateRecord<'a> {
+    fn try_light_accounts(accounts: &'a [LightAccountInfo]) -> Result<Self> {
+        let record: LightAccount<NameRecord> = LightAccount::new(&accounts[0])?;
+        Ok(Self { record })
+    }
+}
+
+#[light_system_accounts]
```

Comment:
(We will still need to validate some accounts ofc. but I think it will make maintenance easier and its probably our biggest anchor dependency.)

Reply:
and we could implement it as follows so that we can use it in the verify function as is: 
it`s not completely thought through rfc :)
```
#[derive(LightSystemAccounts]
struct LightAccountsTraitHelperStruct {
   account_infos: &[&AccountInfos],
}
impl  LightAccountsTraitHelperStruct {
    fn from_remaining_accounts(accounts: &[AccountInfo],start_index: usize) -> Self {
     LightAccountsTraitHelperStruct {
         account_infos: accounts[start_index..NUMBER_OF_SYSTEM_ACCOUNTS],
     }
}
```

Usage:
```
LightAccountsTraitHelperStruct.get_light_system_program();
```

Reply:
Good idea! Will definitely do in the Anchor-less PR.